src/test/scala/is/hail/io/ImportGDBSuite.scala

```diff
@@ -0,0 +1,140 @@
+    val gdbIterator = new GenomicsDBFeatureReader(loader, workspace, arrName, ref, codec).iterator
+
+    var count = 0
+    while (gdbIterator.hasNext) {
+      val record = gdbIterator.next
+      if (!record.getAlternateAlleles.get(0).toString.equals("*")) {
+        count += 1
+      }
+    }
+
+    gdbIterator.close()
+
+    assert(count==numVariants)
+  }
+
+  @Test def genomicsDBGlobalAnnotation() {
+
+    val vcfGA = vcfVariantSampleMatrix.globalAnnotation
+    val gdbGA = gdbVariantSampleMatrix.globalAnnotation
+
+    assert(vcfGA == gdbGA)
+  }
+
+  @Test def genomicsDBNoMulti() {
+
+    assert(vcfVariantSampleMatrix.wasSplit == gdbVariantSampleMatrix.wasSplit) // wasSplit == noMulti
+  }
+
+  @Test def genomicsDBVariantAnnotationSignatures() {
+
+    val vcfVAS = vcfVariantSampleMatrix
+      .metadata
+      .vaSignature
+      .asInstanceOf[TStruct]
+      .fields
+
+    val gdbVAS = gdbVariantSampleMatrix
+      .metadata
+      .vaSignature
+      .asInstanceOf[TStruct]
+      .fields
+
+    assert(vcfVAS.zip(gdbVAS).forall( { case (f1, f2) => f1.equals(f2) }))
+  }
+
+  @Test def genomicsDBGenotypeSignature() {
+
+    val vcfGenotypeSignature = vcfVariantSampleMatrix
+      .genotypeSignature
+      .asInstanceOf[TStruct]
+      .fields
+
+    val gdbGenotypeSignature = gdbVariantSampleMatrix
+      .genotypeSignature
+      .asInstanceOf[TStruct]
+      .fields
+
+    assert(vcfGenotypeSignature.zip(gdbGenotypeSignature).forall( { case (f1, f2) => f1.equals(f2) }))
+  }
+
+  @Test def genomicsDBVariantsSamples() {
+
+    assert(vcfVariantSampleMatrix.count == gdbVariantSampleMatrix.count) //(numSamples, numVariants)
+
+    assert(vcfVariantSampleMatrix.variants.collect.sameElements(gdbVariantSampleMatrix.variants.collect))
+
+    assert(vcfVariantSampleMatrix.sampleIds.equals(gdbVariantSampleMatrix.sampleIds))
+  }
+
+  @Test def genomicsDBRDDGenotypeData() {
+
+    val vcfGT = vcfVariantSampleMatrix
+      .rdd
+      .map(_._2._2)
+      .collect
+
+    val gdbGT = gdbVariantSampleMatrix
+      .rdd
+      .map(_._2._2)
+      .collect
+
+    assert(vcfGT.zip(gdbGT).forall( { case (it1, it2) => it1.zip(it2).forall( { case (a1, a2) => a1.equals(a2) }) }))
+  }
+
+  //FIXME: this test fails because some FP numbers are off by a little, arrays (length "A") dropping all but first element
+  @Test def genomicsDBRDDAnnotations() {
+
+    val vcfAnn = vcfVariantSampleMatrix
+      .rdd
+      .map(_._2._1.asInstanceOf[GenericRow].toSeq(3)) //map out the QUAL field, which is set to missing in GenomicsDB
+      .map(ann => ann.asInstanceOf[GenericRow].toSeq(2)) //TODO: this line maps to a field in the annotation that isn't affected by either bug (above) -- remove this map from tests after genomicsDB fixes issues
+      .collect
+
+    val gdbAnn = gdbVariantSampleMatrix
+      .rdd
+      .map(_._2._1.asInstanceOf[GenericRow].toSeq(3))
+      .map(ann => ann.asInstanceOf[GenericRow].toSeq(2))
```

Comment:
@danking when I map like this instead of casting to a generic row, I get the following error:

> org.apache.spark.SparkException: Task not serializable
> 
> 	at org.apache.spark.util.ClosureCleaner$.ensureSerializable(ClosureCleaner.scala:298)
> 	at org.apache.spark.util.ClosureCleaner$.org$apache$spark$util$ClosureCleaner$$clean(ClosureCleaner.scala:288)
> 	at org.apache.spark.util.ClosureCleaner$.clean(ClosureCleaner.scala:108)
> 	at org.apache.spark.SparkContext.clean(SparkContext.scala:2039)
> 	at org.apache.spark.rdd.RDD$$anonfun$map$1.apply(RDD.scala:366)
> 	at org.apache.spark.rdd.RDD$$anonfun$map$1.apply(RDD.scala:365)
> 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
> 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
> 	at org.apache.spark.rdd.RDD.withScope(RDD.scala:358)
> 	at org.apache.spark.rdd.RDD.map(RDD.scala:365)
> 	at is.hail.io.ImportGDBSuite.genomicsDBRDDAnnotations(ImportGDBSuite.scala:129)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
> 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:696)
> 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:882)
> 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1189)
> 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:124)
> 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
> 	at org.testng.TestRunner.privateRun(TestRunner.java:767)
> 	at org.testng.TestRunner.run(TestRunner.java:617)
> 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
> 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
> 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
> 	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
> 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
> 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
> 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
> 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
> 	at org.testng.TestNG.run(TestNG.java:1057)
> 	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:72)
> 	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:123)
> Caused by: java.io.NotSerializableException: is.hail.variant.VariantSampleMatrix